packages/fxa-auth-server/test/local/payments/stripe.js

```diff
@@ -145,6 +144,91 @@ describe('StripeHelper', () => {
     sandbox.restore();
   });
 
+  describe('getCachedResult', () => {
+    const expectedResult = { status: 'success' };
+
+    const cacheKey = 'testKey';
+    let refreshFunction;
+
+    beforeEach(() => {
+      refreshFunction = sandbox.stub().returns(expectedResult);
+    });
+
+    it('follows read-through caching logic', async () => {
+      // On first call: cache miss, call refresh function, cache the result
+      let actualResult = await stripeHelper.getCachedResult(
+        cacheKey,
+        refreshFunction
+      );
+      assert.isTrue(mockRedis.get.calledOnceWith(cacheKey));
+      assert.isTrue(
+        mockRedis.set.calledOnceWith(cacheKey, JSON.stringify(expectedResult))
+      );
+      assert.isTrue(refreshFunction.calledOnce);
+      assert.deepEqual(actualResult, expectedResult);
+
+      refreshFunction.resetHistory();
+      mockRedis.get.resetHistory();
+      mockRedis.set.resetHistory();
+
+      // On subsequent call: cache hit, return cache result
+      actualResult = await stripeHelper.getCachedResult(
+        cacheKey,
+        refreshFunction
+      );
+      assert.isTrue(mockRedis.get.calledOnceWith(cacheKey));
+      assert.isTrue(mockRedis.set.notCalled);
+      assert.isTrue(refreshFunction.notCalled);
+      assert.deepEqual(actualResult, expectedResult);
+    });
+
+    describe('when there is an error on cache read', () => {
+      it('logs error and returns expected result', async () => {
+        mockRedis.get.restore();
+        sandbox.stub(mockRedis, 'get').rejects(Error);
+
+        const actualResult = await stripeHelper.getCachedResult(
+          cacheKey,
+          refreshFunction
+        );
+
+        assert.isTrue(mockRedis.get.calledOnceWith(cacheKey));
+        assert.isTrue(
+          mockRedis.set.calledOnceWith(cacheKey, JSON.stringify(expectedResult))
```

Comment:
:( I'm not as familiar with the Sinon API as I thought.  `calledOnceWith` only partially matches the args you passed it.  I think we should use `calledOnceWithExactly`.